File: output/x_ai_grok_code_fast_1/few/cpp/Letras.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    string s;
    cin >> s;
    vector<int> dp(26, 0);
    for (char ch : s) {
        int idx = ch - 'A';
        int max_prev = 0;
        for (int i = 0; i <= idx; i++) {
            max_prev = max(max_prev, dp[i]);
        }
        dp[idx] = max(dp[idx], 1 + max_prev);
    }
    int ans = 0;
    for (int x : dp) ans = max(ans, x);
    cout << ans << endl;
    return 0;
}

File: output/gemini_3_1_flash_lite_preview/zero/python_noimg/Letras.py
import sys

def solve():
    s = sys.stdin.readline().strip()
    if not s:
        print(0)
        return
    
    dp = [0] * 26
    
    for char in s:
        idx = ord(char) - ord('A')
        max_prev = 0
        for i in range(idx + 1):
            if dp[i] > max_prev:
                max_prev = dp[i]
        dp[idx] = max_prev + 1
        
    print(max(dp))

if __name__ == '__main__':
    solve()